### PR TITLE
Sendable diagnostics

### PR DESCRIPTION
**Pull request by DougGregor**
Improve sendable diagnostics to distinguish the various reasons why we are performing sendable checking, e.g., uses of an actor-isolated declaration from outside the actor, protocol conformance, `@objc`, and so on.

Cleanup to make rdar://80424675 more useful

### CODE REVIEW

**Comment by DougGregor** [comment]
@swift-ci please smoke test

**Comment by DougGregor** [comment]
@swift-ci please smoke test

File: test/Concurrency/actor_call_implicitly_async.swift [hunk 1]
```diff
@@ -189,8 +189,8 @@ func someAsyncFunc() async {
   _ = a.effPropAT
 
   // (mostly) corrected ones
-  _ = await a.effPropA  // expected-warning {{cannot use property 'effPropA' with a non-sendable type 'Box' across actors}}
-  _ = try! await a.effPropT // expected-warning {{cannot use property 'effPropT' with a non-sendable type 'Box' across actors}}
+  _ = await a.effPropA  // expected-warning {{non-sendable type 'Box' in asynchronous access to actor-isolated property 'effPropA' cannot cross actor boundary}}
```
Comment (ktoso) [inline test/Concurrency/actor_call_implicitly_async.swift:192]: much better, yay!

**Review by ktoso — APPROVED** [review]
Great improvements! LGTM

**Pull request merged**